### PR TITLE
CHK-328: Add handler for cart event

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What is the purpose of this pull request?

Adds the handler for `cart` which will emit an event useful for the checkout funnel report of Google Analytics.

#### What problem is this solving?

Before this we couldn't measure the cart step of the purchase funnel with Checkout IO.

#### How should this be manually tested?

Access [this workspace](https://lucas--checkoutio.myvtex.com/cart) and check the `window.dataLayer`.

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [x] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/typings/events.d.ts [hunk 1]
```diff
@@ -140,21 +141,112 @@ export interface ProductImpressionData extends EventData {
   list: string
 }
 
+export interface CartData extends EventData {
+  event: 'cart'
+  eventName: 'vtex:cart'
+  orderForm: OrderForm
```
Comment (klzns) [inline react/typings/events.d.ts:147]: Can you also add this new event to 
https://github.com/vtex-apps/pixel-app-template

File: react/typings/events.d.ts [hunk 2]
```diff
@@ -140,21 +141,112 @@ export interface ProductImpressionData extends EventData {
...
+
+export interface AnalyticsEcommerceProduct {
+  id: string
+  name: string
+  category: string
+  brand: string
+  variant: string
+  price: number
+  quantity: number
+}
```
Comment (klzns) [inline react/typings/events.d.ts:158]: I would move this to another file (related to Google), the intention of this file is to gather only the pixel events.

**Review by klzns — CHANGES_REQUESTED** [review]

File: react/typings/events.d.ts (hunk 1 above)
Comment (lucasecdb) [reply react/typings/events.d.ts:147]: sure

Comment (lucasecdb) [reply react/typings/events.d.ts:147]: done in vtex-apps/pixel-app-template#13

**Review by klzns — APPROVED** [review]

**Review by darlenedms — APPROVED** [review]

**Pull request merged**